File: frontend/src/api/task.ts
import { apiRequest } from "./client";

export type Task = {
    id: number;
    name: string;
    completed: boolean;
    created_at: string;
    expires_in: string;
    user_id: number;
    expired: boolean;
}

export type PaginatedTasksResponse = {
    tasks: Task[];
    total: number;
    pages: number;
    current_page: number;
}

export const getTasks = (userId: number, token: string, page = 1, perPage = 10) =>
    apiRequest<PaginatedTasksResponse>(`/tasks/user/${userId}?page=${page}&per_page=${perPage}`, {
        method: 'GET',
        token
    })

File: app/models/task.py
from app import db
from datetime import datetime

class Task(db.Model):
    __tablename__ = 'tasks'

    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(80), nullable=False)
    completed = db.Column(db.Boolean, default=True, nullable=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    expires_in = db.Column(db.DateTime, nullable=False)
    expired = db.Column(db.Boolean, default=False, nullable=False)

    user_id = db.Column(
        db.Integer, 
        db.ForeignKey('users.id', ondelete='CASCADE'),
        nullable=False
    )

    user = db.relationship(
        'User',
        back_populates='tasks'
    )

    def serialize(self):
        return {
            'id': self.id,
            'name': self.name,
            'completed': self.completed,
            'created_at': self.created_at.isoformat(),
            'expires_in': self.expires_in.isoformat(),
            'expired': self.expired,
            'user_id': self.user_id
        }